demos/link-prediction/random-walks/ctdne-link-prediction.ipynb

```diff
@@ -0,0 +1,680 @@
+    "print()\n",
+    "print(f\"Training examples: {len(pos)} positive links, {len(neg)} negative links\")\n",
+    "print(\n",
+    "    f\"Validation examples: {len(pos_test)} positive links, {len(neg_test)} negative links\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Running random walks\n",
+    "\n",
+    "We try to keep the setup comparable between the use of temporal and biased (static) random walks. For temporal walks, the input parameter is defined in terms of the total number of context windows you are interested in obtaining, which differs from the traditional approach of specifying the number of walks to run per node in the graph. We calculate the number of context windows we need in terms of the traditional parameters as:\n",
+    "```\n",
+    "num_cw = num_nodes * num_walks_per_node * (walk_length - cw_size)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of temporal random walks: 1763\n",
+      "Number of static random walks: 1510\n"
+     ]
+    }
+   ],
+   "source": [
+    "from stellargraph.data import TemporalRandomWalk, BiasedRandomWalk\n",
+    "\n",
+    "num_walks_per_node = 10\n",
+    "walk_length = 80\n",
+    "cw_size = 10\n",
+    "num_cw = (\n",
+    "    len(graph.nodes()) * num_walks_per_node * (walk_length - cw_size)\n",
+    ")  # as recommended in reference paper\n",
+    "\n",
+    "temporal_rw = TemporalRandomWalk(graph)\n",
+    "temporal_walks = temporal_rw.run(\n",
+    "    num_cw=num_cw,\n",
+    "    cw_size=cw_size,\n",
+    "    max_walk_length=walk_length,\n",
+    "    initial_edge_bias=None,\n",
+    "    walk_bias=\"exponential\",\n",
+    "    p_walk_success_threshold=0.2,\n",
+    ")\n",
+    "\n",
+    "static_rw = BiasedRandomWalk(graph)\n",
+    "static_walks = static_rw.run(\n",
+    "    nodes=graph.nodes(), n=num_walks_per_node, length=walk_length\n",
+    ")\n",
+    "\n",
+    "\n",
+    "print(\"Number of temporal random walks: {}\".format(len(temporal_walks)))\n",
+    "print(\"Number of static random walks: {}\".format(len(static_walks)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from gensim.models import Word2Vec\n",
+    "\n",
+    "temporal_model = Word2Vec(\n",
+    "    temporal_walks, size=128, window=10, min_count=0, sg=1, workers=2, iter=1\n",
+    ")\n",
+    "static_model = Word2Vec(\n",
+    "    static_walks, size=128, window=10, min_count=0, sg=1, workers=2, iter=1\n",
+    ")\n",
+    "\n",
+    "# using the trained word2vec model, create a function we can use to generate embeddings\n",
+    "def temporal_embedding(u):\n",
+    "    try:\n",
+    "        return temporal_model.wv[u]\n",
+    "    except KeyError:\n",
+    "        return np.zeros(128)\n",
```

Comment:
Instead of hardcoding the 128 here, maybe this could be either computed from `temporal_model` (maybe there's a `temporal_model.size` property?) or pulled out into a parameter?

```python
from genesis.models import Word2Vec

embedding_size = 128
temporal_model = Word2Vec(...., size=embedding_size, ...)
static_model = Word2Vec(...., size=embedding_size, ...)

unseen_node_embedding = np.zeros(embedding_size)

def temporal_embedding(u):
    try:
        return temporal_model.wv[u]
    except KeyError:
        return unseen_node_embedding
```